fe2-android/app/src/main/java/com/github/dedis/popstellar/ui/detail/LaoDetailViewModel.java

```diff
@@ -925,27 +754,46 @@ public boolean addManually(String data) {
    * @return true if an attendee was added false otherwise
    */
   private boolean handleAttendeeAddition(String data) {
-    PublicKey attendee;
+    PublicKey publicKey;
     try {
-      attendee = new PublicKey(data);
+      publicKey = new PublicKey(data);
     } catch (IllegalArgumentException e) {
       mScanWarningEvent.postValue(new SingleEvent<>("Invalid key format code. Please try again."));
       return false;
     }
-    if (attendees.contains(attendee) || witnesses.contains(attendee)) {
+    if (attendees.contains(publicKey) || witnesses.contains(publicKey)) {
       mScanWarningEvent.postValue(
           new SingleEvent<>("This attendee key has already been scanned. Please try again."));
       return false;
     }
     if (scanningAction == ScanningAction.ADD_ROLL_CALL_ATTENDEE) {
-      attendees.add(attendee);
+      attendees.add(publicKey);
       mAttendeeScanConfirmEvent.postValue(new SingleEvent<>("Attendee has been added."));
       mNbAttendees.postValue(attendees.size());
     } else if (scanningAction == ScanningAction.ADD_WITNESS) {
-      witnesses.add(attendee);
+      witnesses.add(publicKey);
       mWitnessScanConfirmEvent.postValue(new SingleEvent<>(true));
-      updateLaoWitnesses();
+      disposables.add(
+          updateLaoWitnesses()
+              .subscribe(
+                  () -> Log.d(TAG, "Witness " + publicKey + " added"),
+                  error ->
+                      ErrorUtils.logAndShow(
+                          getApplication(), TAG, error, R.string.error_update_lao)));
     }
     return true;
   }
+
+  /**
+   * This function should be used to add disposable object generated from subscription to sent
+   * messages flows
+   *
+   * <p>They will be disposed of when the view model is cleaned which ensures that the subscription
+   * stays relevant throughout the whole lifecycle of the activity and it is not bound to a fragment
+   *
+   * @param disposable to add
+   */
+  public void addDisposable(Disposable disposable) {
```

Comment:
I see that it is present in all our viewModels. What do you think of defining an interface ?

Reply:
We can't have fields in interfaces.

As such, this would not be very useful, the interface would not do much in that regard.